manual/embedded-examples/src/docs/dev/procedures.asciidoc

```diff
@@ -0,0 +1,495 @@
+Finding dense nodes is a graph-global operation and a good fit for procedural code, to be called from Cypher.
+
+Continuing with the `CALL` example, a procedure to locate dense nodes might take a `threshold` argument, defining how many relationships are necessary for a node to be considered dense.
+
+.Find dense nodes procedure
+[snippet, java]
+----
+component=neo4j-examples
+source=org/neo4j/examples/ProcedureExample.java
+tag=procedureExample
+classifier=sources
+----
+
+The procedure, if it returns anything, will return a `java.util.stream.Stream` of Records.
+
+.Find dense nodes return record
+[snippet, java]
+----
+component=neo4j-examples
+source=org/neo4j/examples/ProcedureExample.java
+tag=outputRecordExample
+classifier=sources
+----
+
+Read more about creating a procedure in <<writing-procedure, Writing a procedure>>.
+
+
+== Built-in procedures
+
+Neo4j comes with a handful of pre-packaged procedures ready to be used:
+
+[options="header", cols="m,m,d"]
+|===
+| Procedure name        | Command to invoke procedure     | What it does
+| ListProcedures        | CALL sys.db.procedures()        | List all procedures in the database.
+| ListLabels            | CALL sys.db.labels()            | List all labels in the database.
+| ListRelationshipTypes | CALL sys.db.relationshipTypes() | List all relationship types in the database.
+| ListPropertyKeys      | CALL sys.db.propertyKeys()      | List all property keys in the database.
+|===
+
+
+== Developing Compiled Procedures
+
+[TIP]
+--
+The example discussed below is available as {procedure-template-url}[a repository on Github].
+To get started quickly you can fork the repository and work with the code as you follow along in the guide below.
+--
+
+Compiled procedures are written in the Java programming language.
+While you can develop Compiled Procedures using a regular text editor it is recommended to use a dedicated Java development environment, like IntelliJ IDEA or Eclipse.
+
+To develop a new Procedure, you need to produce a `jar` file that contains the procedure along with any non-Neo4j dependencies it has.
+This file is then deployed in the `plugin` directory of each database in your cluster.
+The procedure will become available in each instance after restart.
+
+The general process for building a new Compiled Procedure is:
+
+* Set up a new Java project in your IDE
+* Write a test for the Procedure using Neo4j Harness
+* Write a procedure that makes your test pass
+* Build a `jar`-file from your project and deploy it
+
+
+=== Setting up a new project
+
+You can set up your project in any way that allows you to produce a Jar file.
+Below is an example of how to configure the Maven build system to compile a Procedure and produce a `jar` file.
+The Maven _.pom_-file is broken into pieces with comments for readability, the whole file is available from the {procedure-template-url}[Neo4j Procedure Template] repository.
```

Comment:
_.pom_file: pom isn't the extension :-)
In line you 131 you refer to the same file differently: _pom.xml_